### PR TITLE
Remove "send enabled" on marker removal and in bulk on 1.10.0 upgrade

### PR DESCRIPTION
**Pull request by derekadams**
## Description

Remove the "send enabled" bank denom metadata in cases where a marker is deleted. Also add an upgrade handler (`lava`) that iterates through the "send enabled" metadata and removes it for any that is not associated with a marker.

closes: #819

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by Taztingo** [comment]
It looks like linting is failing. Think you just have to run **make lint**

Here is the full documentation on it...

Please make sure to run **make format** before every commit - the easiest way to do this is have your editor run it for you upon saving a file. Additionally please ensure that your code is lint compliant by running **make lint**. A convenience git pre-commit hook that runs the formatters automatically before each commit is available in the contrib/githooks/ directory.

**Comment by channa-figure** [comment]
Is it possible to have a test for this?  Or is that not needed?

File: app/upgrades.go [hunk 1]
```diff
@@ -96,7 +96,13 @@ var handlers = map[string]appUpgrade{
 		}, // upgrade for pio-testnet-1 from v1.8.0-rc8 to v1.8.0-rc9
 	},
 	"kahlua": {}, // upgrade for pio-testnet-1 from v1.8.0-rc9 to v1.8.0
-	"lava":   {}, // 1.9.0
+	"lava": {
+		Handler: func(app *App, ctx sdk.Context, plan upgradetypes.Plan) (module.VersionMap, error) {
+			app.MarkerKeeper.RemoveSendEnabledForMissingMarkers(ctx)
+			versionMap := app.UpgradeKeeper.GetModuleVersionMap(ctx)
+			return versionMap, nil
+		}, // upgrade for 1.9.1
```
Comment (iramiller) [inline app/upgrades.go:104]: might not be 1.9.1 anymore ... but we can tweak this in the release branch this ends up in.

**Review by iramiller — APPROVED** [review]
LGTM

**Comment by iramiller** [comment]
This could stand to have a simple test added for the removeSendEnabled flow...

**Comment by channa-figure** [comment]
Agree with @iramiller could add the keeper test.  I don't think we have a test flow for the upgrade handlers.

**Review by channa-figure — DISMISSED** [review]
Thanks for the test.  LGTM.

File: app/upgrades.go (hunk 1 above)
Comment (derekadams) [reply app/upgrades.go]: yep, updated to 1.10.0

**Review by channa-figure — APPROVED** [review]

**Comment by iramiller** [comment]
Test should have a denom that is valid (hash?) which is not removed ... that would show both code paths ... otherwise looks good.

**Review by iramiller — APPROVED** [review]

**Review by channa-figure — APPROVED** [review]

**Pull request merged**